src/Propulsion.Cosmos/ChangeFeedProcessor.fs

```diff
@@ -128,12 +133,15 @@ type ChangeFeedProcessor =
             leasePrefix |> Option.iter (fun lp -> feedProcessorOptions.LeasePrefix <- lp + ":")
             // Max Items is not emphasized as a control mechanism as it can only be used meaningfully when events are highly regular in size
             maxDocuments |> Option.iter (fun mi -> feedProcessorOptions.MaxItemCount <- Nullable mi)
-            let mk (Equinox.Cosmos.Discovery.UriAndKey (u,k)) d c =
-                DocumentCollectionInfo(Uri = u, DatabaseName = d, CollectionName = c, MasterKey = k, ConnectionPolicy = connectionPolicy)
+            let mkC = function
+                | Discovery.UriKeyAndPolicy (u,k,p) -> new DocumentClient(serviceEndpoint=u, authKeyOrResourceToken=k, connectionPolicy=p,desiredConsistencyLevel=Nullable())
+                | Discovery.Custom documentClient -> documentClient
             ChangeFeedProcessorBuilder()
                 .WithHostName(leaseOwnerId)
-                .WithFeedCollection(mk discovery source.database source.container)
-                .WithLeaseCollection(mk (defaultArg auxDiscovery discovery) aux.database aux.container)
+                .WithFeedCollection(DocumentCollectionInfo(DatabaseName=source.database, CollectionName=source.container))
+                .WithLeaseCollection(DocumentCollectionInfo(DatabaseName=aux.database, CollectionName=aux.container))
```

Comment:
It looks like `builder.BuildEstimatorAsync()` below will throw a `System.NullReferenceException` if the `DocumentCollectionInfo.Uri` property isn't given to `.WithFeedCollection(..)`. Seems to work without explicitly specifying the service endpoint in `.WithLeaseCollection(..)` though.

We call `mkC` twice here -- if the UriAndKey discovery is used, with no auxDiscovery specified, do we risk creating two identical DocumentClients? I'm not sure if the CFPBuilder had logic to prevent this from happening, or if this is already the current behavior.

Reply:
Looking in the source, I can see why that is; I've amended the inputs to require the Uri too. Also, the `ConnectionPolicy` is used in other contexts too so have required that.
The existing logic already resulted in two DocumentClients being created from the discovery and connectionPolicy; I assume the DocDb client wiring or the CFP connection logic collapses those - the effective outcome seems the same as it is now